### PR TITLE
Pgcraft misc

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: doc/usr/expl/shaders.md [hunk 1]
```diff
@@ -160,11 +160,11 @@ To sample a `Texture2D` from within the shader, multiple options are available:
...
 
-Type   | Name              | Description
--------|-------------------|------------
-`mat4` | `%s_coord_matrix` | uv transformation matrix of the texture associated with the render node using key `%s`, it should be applied to the geometry uv coordinates `ngl_uvcoord` to obtain the final texture coordinates. This variable is always exposed in the `vertex` stage.
-`vec2` | `%s_dimensions`   | dimensions in pixels of the texture associated with the render node using key `%s`. This variable is exposed in the same stage as the texture.
-`float`| `%s_ts`           | timestamp generated by the texture data source, 0.0f for images and buffers, frame timestamp for audios and videos. This variable is exposed in the same stage as the texture.
+Availability | Type   | Name              | Description
+-------------|-------|-------------------|------------
+Texture*     |`mat4` | `%s_coord_matrix` | uv transformation matrix of the texture associated with the render node using key `%s`, it should be applied to the geometry uv coordinates `ngl_uvcoord` to obtain the final texture coordinates. This variable is always exposed in the `vertex` stage.
```
Comment (ubitux) [inline doc/usr/expl/shaders.md:165]: Please use backticks for content in this column

File: libnopegl/src/pgcraft.c [hunk 2]
```diff
@@ -446,7 +439,7 @@ static int prepare_texture_infos(struct pgcraft *s, const struct pgcraft_params
     return 0;
 }
 
-static int inject_texture_info(struct pgcraft *s, struct pgcraft_texture_info *info, int stage)
+static int inject_texture(struct pgcraft *s, struct pgcraft_texture *texture, struct pgcraft_texture_info *info, int stage)
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:433]: I think `texture` can be `const`

File: libnopegl/src/pgcraft.c [hunk 3]
```diff
@@ -523,10 +523,9 @@ static int inject_texture(struct pgcraft *s, struct pgcraft_texture *texture, st
 
 static int inject_textures(struct pgcraft *s, const struct pgcraft_params *params, int stage)
 {
-    struct darray *texture_infos_array = &s->texture_infos;
     struct pgcraft_texture *textures = ngli_darray_data(&s->textures);
-    struct pgcraft_texture_info *texture_infos = ngli_darray_data(texture_infos_array);
-    for (size_t i = 0; i < ngli_darray_count(texture_infos_array); i++) {
+    struct pgcraft_texture_info *texture_infos = ngli_darray_data(&s->texture_infos);
+    for (size_t i = 0; i < ngli_darray_count(&s->texture_infos); i++) {
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:526]: commit message is missing a `s` in "simplify inject_texture**s**()"

File: libnopegl/src/pgcraft.c [hunk 4]
```diff
@@ -1231,26 +1234,33 @@ static int32_t get_texture_index(const struct pgcraft *s, const char *name)
     return -1;
 }
 
-static void probe_texture_info_elems(const struct pgcraft *s, struct pgcraft_texture_info_field *fields)
+static void probe_texture_info_elems(const struct pgcraft *s,
+                                     struct pgcraft_texture *texture,
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:1233]: Again here the added `texture` can be `const`.

File: libnopegl/src/pgcraft.c [hunk 5]
```diff
@@ -1231,26 +1234,33 @@ static int32_t get_texture_index(const struct pgcraft *s, const char *name)
...
+            field->index = get_texture_index(s, name);
         else
-            field->index = ngli_pgcraft_get_uniform_index(s, field->name, field->stage);
+            field->index = ngli_pgcraft_get_uniform_index(s, name, field->stage);
     }
 }
 
 static void probe_texture_infos(struct pgcraft *s)
 {
+    struct pgcraft_texture *textures = ngli_darray_data(&s->textures);
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:1252]: `const` here too

File: libnopegl/src/pgcraft.c [hunk 6]
```diff
@@ -1255,9 +1255,8 @@ static void probe_texture_info_elems(const struct pgcraft *s,
 static void probe_texture_infos(struct pgcraft *s)
 {
     struct pgcraft_texture *textures = ngli_darray_data(&s->textures);
-    struct darray *texture_infos_array = &s->texture_infos;
-    struct pgcraft_texture_info *texture_infos = ngli_darray_data(texture_infos_array);
-    for (size_t i = 0; i < ngli_darray_count(texture_infos_array); i++) {
+    struct pgcraft_texture_info *texture_infos = ngli_darray_data(&s->texture_infos);
+    for (size_t i = 0; i < ngli_darray_count(&s->texture_infos); i++) {
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:1254]: Incorrect commit message; it's not `probe_texture_info_elems()` but `probe_texture_infos()`

File: CHANGELOG.md [hunk 7]
```diff
@@ -22,6 +22,10 @@ Versioning](https://semver.org/spec/v2.0.0.html) for `libnopegl`.
   v2023.2, this is changed back in this release
 - `SceneCfg` and `SceneInfo` now use the `Backend` enum instead of a string
 
+### Removed
+- `%s_dimensions` uniform for 2D array and 3D images/textures, users must use
+  textureSize()/imageSize() instead
```
Comment (ubitux) [inline CHANGELOG.md:27]: Please state that it is still available for 2D textures, whether they're accelerated or not

**Review by ubitux — CHANGES_REQUESTED** [review]

File: libnopegl/src/pgcraft.c (hunk 3 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c:526]: Fixed.

File: libnopegl/src/pgcraft.c (hunk 4 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c]: Done.

File: libnopegl/src/pgcraft.c (hunk 5 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c]: Done.

File: libnopegl/src/pgcraft.c (hunk 6 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c:1254]: Fixed.

File: CHANGELOG.md (hunk 7 above)
Comment (mbouron) [reply CHANGELOG.md]: Done.

File: libnopegl/src/pgcraft.c (hunk 2 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c]: Done.

**Review by ubitux — APPROVED** [review]

**Pull request merged**